deployment/terraform/engine.go

```diff
@@ -102,10 +102,14 @@ func _runCommand(cmd *exec.Cmd, dst io.Writer) error {
 		}
 	}()
 
-	scanner := bufio.NewScanner(stdout)
-	for scanner.Scan() {
-		mlog.Info(scanner.Text())
-	}
+	go func() {
```

Comment:
What is the benefit of doing this? Scanning stdout is not really blocking anything. And anyways you need to wait until stderr scanning is finished.

Reply:
Without this I was not getting the entire output from terraform, some details about errors where missing.

Reply:
Ok, that's weird. I don't understand how this changes anything though. Probably we can look into this in an isolated scenario. FYI @agarciamontoro @streamer45

Reply:
Hmm .. ok I understand the bug. https://pkg.go.dev/os/exec#Cmd.StdoutPipe

> [Cmd.Wait](https://pkg.go.dev/os/exec#Cmd.Wait) will close the pipe after seeing the command exit, so most callers need not close the pipe themselves. It is thus incorrect to call Wait before all reads from the pipe have completed.

So using a scanner to read from the pipe is error-prone and probably why it fails a lot of times. The better way is to directly set the StdErr/Stdout fields with a custom io.Writer.

```go
type cmdLogger struct {
}

func (*cmdLogger) Write(in []byte) (int, error) {
	mlog.Info(string(in))
	return len(in), nil
}
```

and then 

```go
cmd.Stdout = &cmdLogger
cmd.Stderr = cmd.Stdout

if err := cmd.Start(); err != nil {
return err
}

if err := cmd.Wait(); err != nil {
return err
}
```

Having the same writer for stdout and stderr is handled properly in the stdlib:
> // If Stdout and Stderr are the same writer, and have a type that can
	// be compared with ==, at most one goroutine at a time will call Write.

@fmartingr - can we try with something like this?